### ISSUE TITLE
get a11y feedback for fx56

### ISSUE THREAD

**Issue by dmose** [mozilla/activity-stream#2874]
We'll need to work with Marco Zehe or Yura to get feedback about what changes we're still going to need in order for this to ride the 56 trains.

**Comment by yzen** [comment]
Just a quick observation import dialog (underneath search) has no keyboard focus styling. Buttons are actionable but there is not way to see what button is being activated as focus is not visible. Note: hover styling is there.

**Comment by yzen** [comment]
Overall it is pretty good, here are more comments to make it awesome (in addition to the one from yesterday):
- Potential semantics improvements:
  - .manual-migration-container would ideally be a section (like top sites), this way screen readers can jump between sections (right now it's just a div). It would be even better if it had an aria-label describing the section.
  - top sites section should ideally have aria-labelledby attribute pointing to the h3.section-title (but via id)
  - h3 seems to be a hierarchically invalid header (no preceding h1/h2 that I can find)
-  .context-menu-button:
   - main issue is that it's a hover only element. It's not discoverable unless keyboard focus shifts to it from the link. I would suggest making it visible when focus is on the top site so there is no surprise navigation to a suddenly visible element. I don't think there's no focus specific styling for the button at the moment.
   - .sr-only span inside (afaik used only for text) could be replaced with aria-label or title attribute on the button itself 
   - button should make use of aria-haspopup, aria-controls and aria-expanded attributes
   - looks like when activated focus styling moves onto the preview link
   - when tab away from the menu focus indication disappears, though it is probably still on the button that is now hidden
   - arrow up/down is also expected for menus like that
 - Pocket section
   - section should ideally have aria-labelledby attribute pointing to the h3.section-title (but via id)
   - section-info-option is not accessible (for example screen/keyboard reader user will never be able to send feedback): it has to be in tab index, it has also no semantic information (could be role="note"), sr-only could be eliminated in favour title or aria-label. it should have aria-haspopup, aria-controls and aria-expanded attributes (I would expand it as soon as it's focused). you can probably use aria-describedby attribute here to point at the info-option header and body. info-option-header should have header semantics.

**Comment by Mardak** [comment]
@yzen for those issues, how important are they for fixing in Firefox 56 where only Nightly 56 and ~10% of Beta 56 and 0% of Release 56 will see activity stream?

**Comment by yzen** [comment]
@Mardak, I think 57 is a good target for these (the only content that is not very accessible is the one related to 'section-info-option' if there are any space cycles).

**Comment by k88hudson** [comment]
Thanks @yzen!!
